Client/src/Pages/Status/CreateStatus/index.jsx

```diff
@@ -0,0 +1,97 @@
+import PropTypes from "prop-types";
+import { useNavigate } from "react-router";
+import { useEffect, useState } from "react";
+import { Box, Tab, useTheme } from "@mui/material";
+import TabContext from "@mui/lab/TabContext";
+import TabList from "@mui/lab/TabList";
+import GeneralSettingsPanel from "../../../Components/TabPanels/Status/GeneralSettingsPanel";
+import {
+	capitalizeFirstLetter,
+	toLowerCaseFirstLetter,
+} from "../../../Utils/stringUtils";
+import ContentPanel from "../../../Components/TabPanels/Status/ContentPanel";
+
+/**
+ * CreateStatus page renders a page with tabs for general settings and contents.
+ * @param {string} [props.open] - Specifies the initially open tab: 'general settings' or 'content'.
+ * @returns {JSX.Element}
+ */
+
+const CreateStatus = ({ open = "general-settings" }) => {
+	const theme = useTheme();
+	const navigate = useNavigate();
+	let tabList = ["General Settings", "Contents"];
+	const tab = open
+		.split("-")
+		.map((a) => capitalizeFirstLetter(a))
+		.join(" ");
```

Comment:
We are using two different things to determine the active tab: navigation + the open prop. 
I suggest ditching the open prop, and use the useParams hook or similar to get the url params.
That way we can simply navigate to the route, without the need to parse that data.

Reply:
Hello Caio, although what you suggest sounds like reasonable, I found all of them are needed, just for diffeent cases;  The open prop is for the initial landing, e.g, coming from nested sidebar menu ; the navigation is for when click on the tab to switch between the tabs; the tabIdx state is for showing the actual tab panel; these 3 things look all necessary at the moment as they all take care of different things

Reply:
Yes, I understand. I am suggesting you bing the url parameter to the state. That way you won't need the prop, you will keep the state in the url. 

if the url is status/first-tab you will get the parameter, and you have the state 'first-tab'. That way you can ditch  all of those (props and the functions) and keep just the url and the navigate